### PR TITLE
Rearrage docs

### LINKED ISSUE #522: Separate source docs from bundle format docs
Today in the [plain bundle spec](https://github.com/operator-framework/rukpak/blob/main/docs/plain-bundle-spec.md), we talk about sourcing details unrelated to the actual spec of the plain bundle format. For example, the "Technical Details" section talks about images vs git repos.

I think we've intentionally defined a separation between sourcing a bundle and parsing a bundle that we should reflect in our documentation.

IMO, we should re-organize our docs such that we have something along the lines of:

```console
docs
├── bundles
│   ├── helm.md
│   ├── plain.md
│   └── registry.md
├── provisioners
│   ├── helm.md
│   ├── plain.md
│   └── registry.md
└── sources
    ├── git.md
    ├── http.md
    ├── image.md
    ├── local.md
    └── upload.md
```

Where:
- Bundle specs do not mention source implementations
- Source docs do not mention bundle specs
- Provisioner docs link to sources/bundle docs as necessary to explain what they support.

### PR DESCRIPTION
**Pull request by akihikokuroda**
Closes #522 

Signed-off-by: akihikokuroda <akuroda@us.ibm.com>

### CODE REVIEW

File: docs/bundles/helm.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Helm Bundle Spec
+
+## Overview
+
+This document is meant to define the helm bundle format as a reference for those publishing helm bundles for use with
+RukPak. A bundle is a collection of Kubernetes resources that are packaged together for the purposes of installing onto
+a Kubernetes cluster. A bundle can be unpacked onto a running cluster, where controllers can then create the underlying
+content embedded in the bundle. The bundle can be used as the underlying `spec.source` for
+a [Bundle](https://github.com/operator-framework/rukpak#bundle) resource.
```
Comment (tylerslaton) [inline docs/bundles/helm.md:9]: We repeat this paragraph for every bundle document, perhaps we should update it to just point to the `bundle` section of the root readme instead and say something like:
```suggestion
This document is meant to define the helm bundle format as a reference for those publishing helm bundles for use with RukPak. For more information on the concept of a bundle, click [here](https://github.com/operator-framework/rukpak#bundle).
```
Applies to all subsequently similar entries.

File: docs/provisioners/helm.md [hunk 2]
```diff
@@ -0,0 +1,517 @@
+# helm Provisioner
+
+## Summary
+
+The `helm` provisioner is able to instantiate a given `helm+v3` bundle onto a cluster and then install the referred helm chart
```
Comment (tylerslaton) [inline docs/provisioners/helm.md:5]: Can we start each out these provisioner documents by linking to the root provisioner documentation in someway? Applied to all provisioner docs.

https://github.com/operator-framework/rukpak/tree/main/internal/provisioner

**Review by tylerslaton — COMMENTED** [review]
Had some comments but once those are addressed this looks good to me!

File: docs/provisioners/helm.md (hunk 2 above)
Comment (akihikokuroda) [reply docs/provisioners/helm.md]: I'll update.  Thanks!

File: docs/bundles/helm.md (hunk 1 above)
Comment (akihikokuroda) [reply docs/bundles/helm.md]: I'll update.  Thanks!

**Review by tylerslaton — APPROVED** [review]

**Comment by tylerslaton** [comment]
@joelanford @timflannagan Anything look blocking here for us to merge?

**Review by timflannagan — APPROVED** [review]

**Pull request merged**